eoscompanion/routes.py

```diff
@@ -291,8 +297,13 @@ def companion_app_server_resource_route(server, msg, path, query, *args):
 
     resource_suffix = os.path.splitext(resource_uri)[1]
 
+    if resource_file is None:
+        not_found_response(msg, path)
+        return
+
     # Assume text/html if we don't know the content type for the suffix
     return_content_type = _SUFFIX_CONTENT_TYPES.get(resource_suffix, 'text/html')
+    content_adjuster_cls = _CONTENT_ADJUSTERS.get(query.get('adjuster', None), None)
```

Comment:
How does the requester know that it has to pass the adjuster? It would be better if we could detect that automatically without any special knowledge from the client...

Reply:
In this case the client doesn't *really* have special knowledge.

The `/license` routes are all more or less internal. What happens is that we when we render a page we rewrite all the links. The `adjuster` parameter is included if necessary (for instance, if we are rendering a license file, then all embedded URIs in that license file will be rewritten as `/resource?uri=...adjuster=license` so that we recursively apply the rule. This is completely transparent to the client, which, as far as it is concerned, just has to navigate to that link in its web browser.